lib/ansible/module_utils/ibm_svc_utils.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (C) 2019 IBM CORPORATION
+# Author(s): John Hetherington <john.hetherington@uk.ibm.com>
+#
+# Simplified BSD License (see licenses/simplified_bsd.txt or
+# https://opensource.org/licenses/BSD-2-Clause)
+
+""" Support class for IBM SVC ansible modules """
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+import logging
+
+from ansible.module_utils.basic import jsonify
+from ansible.module_utils.urls import open_url
+from ansible.module_utils.six.moves.urllib.parse import quote
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+
+
+def svc_argument_spec():
+    """
+    Returns argument_spec of options common to ibm_svc_*-modules
+
+    :returns: argument_spec
+    :rtype: dict
+    """
+    return dict(
+        clustername=dict(type='str', required=True),
+        domain=dict(type='str', default=None),
+        validate_certs=dict(type='bool', default=False),
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True),
+        log_path=dict(type='str')
+    )
+
+
+class IBMSVCRestApi(object):
+    """ Communicate with SVC via RestApi
+    SVC commands usually have the format
+    $ command -opt1 value1 -opt2 value2 arg1 arg2 arg3
+    to use the RestApi we transform this into
+    https://host:7443/rest/command/arg1/arg2/arg3
+    data={'opt1':'value1', 'opt2':'value2'}
+    """
+
+    def __init__(self, module, clustername, domain, username, password,
+                 validate_certs, log_path):
+        """ Initialize module with what we need for initial connection
+        :param clustername: name of the SVC cluster
+        :type clustername: string
+        :param domain: domain name to make a fully qualified host name
+        :type domain: string
+        :param username: SVC username
+        :type username: string
+        :param password: Password for user
+        :type password: string
+        :param validate_certs: whether or not the connection is insecure
+        :type validate_certs: bool
+        """
+        self.module = module
+        self.clustername = clustername
+        self.domain = domain
+        self.username = username
+        self.password = password
+        self.validate_certs = validate_certs
+
+        # logging setup
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.debug = self._logger.debug
+        if log_path:
+            logging.basicConfig(level=logging.DEBUG, filename=log_path)
+
+        # Make sure we can connect via the RestApi
+        self.token = self._svc_authorize()
+        self.debug("_connect token=%s", self.token)
```

Comment:
It is better to not to log sensitive information such as tokens.

Reply:
Thanks, fixed it